pagopa/io-app
# feat(Piattaforma Notifiche): [IAMVL-85] Add an internal link for activating PN from a CTA

## Short description
### ⚠️ **This PR depends on #4069**

The PN service must be activated from a CTA included in a generic message. The URL will be:
```ioit://services/service-detail?serviceId={id}&activate=true```

This PR adds the `service-detail` route and a new prop to `ServiceDetailsScreen` for requesting the auto-activation.

https://user-images.githubusercontent.com/467098/180180391-09ca4fef-1816-49b0-b294-9fc32a13e681.mp4

## How to test
Use the [IIP-18-add-pn-endpoints](https://github.com/pagopa/io-dev-api-server/tree/IIP-18-add-pn-endpoints) branch of the dev server and add this to your `config.json`:
```
{
  "services": {
    "includePn": true
  }
}
```
Set `PN_ENABLED=YES` to `.env` and run the app, then open a generic message, you will find a new entry in the internal navigation section. When tapped you should navigate to the PN service screen, sending an activation to the backend automatically.
